app/scripts/views/mixins/notifications-mixin.js

```diff
@@ -0,0 +1,117 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A view mixin to interface with the notifications channel
+ *
+ * Any registered handlers are automatically unregistered
+ * when the view is destroyed.
+ */
+
+define([
+], function () {
+  'use strict';
+
+  var NotificationsChannelMixin = {
+    notificationsInitialize: function (options) {
+      options = options || {};
+
+      /**
+       * A list of handlers registered for this view is kept
+       * so all listeners can be unregistered when the view is
+       * destroyed.
+       */
+      this._notificationsMessages = [];
+      this._notifications = options.notifications;
+
+      for (var eventName in this.notifications) {
+        console.log('registering', eventName);
+        var callback = this.notifications[eventName];
+        if (_.isString(callback)) {
+          callback = (this[callback]).bind(this);
+        }
+
+        this.notificationsOn(eventName, callback);
+      }
+
+      this.on('destroy', this.notificationsOffAll.bind(this));
+    },
+
+    /**
+     * Register a listener
+     *
+     * @method notificationsOn
+     * @param {string} eventName
+     * @param {function} callback
+     */
+    notificationsOn: function (eventName, callback) {
+      this._notifications.on(eventName, callback);
+
+      this._notificationsMessages.push({
+        callback: callback,
+        name: eventName
+      });
+    },
+
+
+    /**
+     * Unregister a listener
+     *
+     * @method notificationsOff
+     * @param {string} eventName
+     * @param {function} callback
+     */
+    notificationsOff: function (eventName, callback) {
+        console.log('unregistering', eventName, callback);
+      this._notificationsMessages.forEach(function (envelope, index) {
+        if (envelope.name === eventName && envelope.callback === callback) {
+          console.log(eventName, 'unregistered');
+          this._notificationsMessages.splice(index, 1);
+          this._notifications.off(envelope.name, envelope.callback);
+        }
+      }, this);
+    },
+
+    /**
+     * Unregister all listeners
+     *
+     * @method notificationsOffAll
+     */
+    notificationsOffAll: function () {
+      console.log('unregistering all');
+      this._notificationsMessages.forEach(function (envelope) {
+        console.log('unregistering', JSON.stringify(envelope));
+        this._notifications.off(envelope.name, envelope.callback);
+      }, this);
+      this._notificationsMessages = [];
+    },
+
+    /**
+     * Send a message over the notifications channel
+     *
+     * @method notify
+     * @param {string} eventName
+     * @param {object} [data]
+     */
+    notify: function (eventName, data) {
+      this._notifications.trigger(eventName, data, this);
+    },
+
+    broadcast: function (eventName, data) {
```

Comment:
...and then `notify` and `broadcast` break the convention set by the other methods.